### DISCUSSION TITLE
Bulk create for realtions

### DISCUSSION THREAD

**Discussion by luisjimenez6245** [collerek/ormar#243] [Q&A]
Is there a way to excuse bulk create in a ManyToMany relation?

I don´t care about retrieving again my info, but it could be cool If it could be a way to insert many items to a relation.

Here´s an example of the code im trying to explain

`
  categories:List[Category]= await Category.objects.all()

  song:Song = await Song.objects.get(id = 0)

  await song.categories.bulk_save(categories)
`

**Comment by collerek** [comment]
If both of your sides are saved (and I assume it's true from your snipper) you can simply manually bulk_create the through model, which is a simple ormar model too.

So if you provided the through model in your ManyToMany declaration you already have it.
If you relied on ormar to create one for you you can get it like this:

```python
ThroughModel = Song.Meta.model_fields["categories"].through
```

Now that you have the through model is as simple as:

```python
categories:List[Category]= await Category.objects.all()
song:Song = await Song.objects.get(id = 0)

# note that this assumes that you didn't customize through relation fields, if applicable change the param names
# you can also simply pas song id (0 in this case) and not fetch the song at all
# if your through model have additional fields you can pass them here too
through_instances = [ThroughModel(song=song.pk, category=category.pk) for category in categories]

# bulk create the through models
await ThroughModel.objects.bulk_create(through_instances)
```
